homeassistant/components/goodwe/button.py

```diff
@@ -0,0 +1,84 @@
+"""GoodWe PV inverter selection settings entities."""
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from datetime import datetime
+import logging
+
+from goodwe import Inverter, InverterError
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, KEY_DEVICE_INFO, KEY_INVERTER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class GoodweButtonEntityDescriptionRequired:
+    """Required attributes of GoodweButtonEntityDescription."""
+
+    action: Callable[[Inverter], Awaitable[None]]
+
+
+@dataclass
+class GoodweButtonEntityDescription(
+    ButtonEntityDescription, GoodweButtonEntityDescriptionRequired
+):
```

Comment:
Why are these split?

Reply:
Because without it pylance would complain with "Fields without default values cannot appear after fields with default values".
It's the same patter of adding non-default/mandatory values as used e.g. in GoodweNumberEntityDescription and I thing also rest of the HA sources.